File: MainApp/src/main/java/com/sleticalboy/dailywork/csv/AutoSwitchUI.java
package com.sleticalboy.dailywork.csv;

import com.sleticalboy.dailywork.R;
import com.sleticalboy.dailywork.base.BaseActivity;
import com.sleticalboy.weight.AutoSwitchView;

import java.util.ArrayList;
import java.util.List;

/**
 * Created on 18-10-23.
 *
 * @author leebin
 */
public final class AutoSwitchUI extends BaseActivity {

    private AutoSwitchView autoSwitchView;

    @Override
    protected int layoutResId() {
        return R.layout.activity_auto_switch;
    }

    @Override
    protected void initView() {
        autoSwitchView = findViewById(R.id.auto_switch_view);
    }

    @Override
    protected void initData() {
        List<String> textList = new ArrayList<>();
        for (int i = 0; i < 31; i++) {
            textList.add("第 " + i + " 条数据");
        }
        autoSwitchView.setTextList(textList);
        autoSwitchView.setAnimTime(300);
        autoSwitchView.setTextStillTime(3000);
    }

    @Override
    protected void onPause() {
        super.onPause();
        autoSwitchView.stop();
    }

    @Override
    protected void onResume() {
        super.onResume();
        autoSwitchView.start();
    }
}

File: MainApp/src/main/java/com/sleticalboy/learning/csv/AutoSwitchUI.kt
package com.sleticalboy.learning.csv

import com.sleticalboy.learning.R
import com.sleticalboy.learning.base.BaseActivity
import com.sleticalboy.weight.AutoSwitchView
import java.util.*


/**
 * Created on 18-10-23.
 *
 * @author leebin
 */
class AutoSwitchUI : BaseActivity() {
    private var autoSwitchView: AutoSwitchView? = null
    override fun layoutResId(): Int {
        return R.layout.activity_auto_switch
    }

    override fun initView() {
        autoSwitchView = findViewById(R.id.auto_switch_view)
    }

    override fun initData() {
        val textList: MutableList<String?> = ArrayList()
        for (i in 0..30) {
            textList.add("第 $i 条数据")
        }
        autoSwitchView!!.setTextList(textList)
        autoSwitchView!!.setAnimTime(300)
        autoSwitchView!!.setTextStillTime(3000)
    }

    override fun onPause() {
        super.onPause()
        autoSwitchView!!.stop()
    }

    override fun onResume() {
        super.onResume()
        autoSwitchView!!.start()
    }
}